src/System.Transactions/src/System/Transactions/TransactionOptions.cs

```diff
@@ -23,17 +23,11 @@ public IsolationLevel IsolationLevel
 
         public override int GetHashCode() => base.GetHashCode();  // Don't have anything better to do.
 
-        public override bool Equals(object obj)
-        {
-            if (!(obj is TransactionOptions))
-            {
-                // Can't use 'as' for a value type
-                return false;
-            }
-            TransactionOptions opts = (TransactionOptions)obj;
-
-            return (opts._timeout == _timeout) && (opts._isolationLevel == _isolationLevel);
-        }
+        public override bool Equals(object obj) => obj is TransactionOptions && Equals((TransactionOptions)obj);
+
+        private bool Equals(TransactionOptions other) =>
```

Comment:
I considered implementing `IEquatable<TransactionOptions>`, but decided not to since this is a public type, even though doing so would technically be "ok" without that yet in the contract. So instead of a public `Equals(TransactionOptions)` or an explicit implementation, I decided to just add a private `Equals(TransactionOptions)` for now.

Would you prefer if this was directly in the `==` operator implementation?